### PR TITLE
fix: deprecate selectedBackgroundColor and selectedBackgroundImage

### PR DESCRIPTION
**Pull request by garymathews**
- Android has logged both `selectedBackgroundColor` and `selectedBackgroundImage` as deprecated for a while in favor of `backgroundSelectedColor` and `backgroundSelectedImage`
- Deprecate the same properties on iOS for parity with Android including `selectedBackgroundGradient`
- Update documentation of deprecated properties

##### TEST CASE
```JS
const win = Ti.UI.createWindow({
    backgroundColor: 'white'
});
const tableViewSection = Ti.UI.createListSection({
    headerTitle: 'TableView'
});
const listViewSection = Ti.UI.createListSection({
    headerTitle: 'ListView'
});
const listView = Ti.UI.createListView({
    separatorStyle: Ti.UI.TABLE_VIEW_SEPARATOR_STYLE_SINGLE_LINE,
    separatorColor: 'black',
    templates: {
        template: {
            properties: {
                layout: 'vertical'
            },
            childTemplates: [{
                    type: 'Ti.UI.Label',
                    bindId: 'title',
                    properties: {
                        top: 5,
                        left: 5,
                        right: 5,
                        font: {
                            fontSize: 18
                        },
                        color: 'black'
                    }
                },
                {
                    type: 'Ti.UI.Label',
                    bindId: 'description',
                    properties: {
                        bottom: 5,
                        left: 5,
                        right: 5,
                        font: {
                            fontSize: 14
                        },
                        color: 'black'
                    }
                },
            ]
        }
    },
    defaultItemTemplate: 'template',
    sections: [tableViewSection, listViewSection]
});
listView.addEventListener('itemclick', e => {
    const item = e.section.items[e.itemIndex];
    if (item.properties.test instanceof Function) {
        item.properties.test();
    }
});
[
    // TableView
    {
        section: tableViewSection,
        title: 'TIMOB-26304',
        description: 'backgroundSelectedColor and backgroundSelectedImage test.',
        test() {
            const win = Ti.UI.createWindow({
                backgroundColor: 'gray'
            });
            const tableView = Ti.UI.createTableView({});
            tableView.setData([{
                    title: 'backgroundSelectedColor',
                    color: 'white',
                    backgroundSelectedColor: 'red'
                },
                {
                    title: 'backgroundSelectedImage',
                    color: 'white',
                    backgroundSelectedImage: 'https://garymathews.com/test.png'
                }
            ]);
            win.add(tableView);
            win.open();
        }
    },

    // ListView
    {
        section: listViewSection,
        title: 'TIMOB-26304',
        description: 'backgroundSelectedColor and backgroundSelectedImage test.',
        test() {
            const win = Ti.UI.createWindow({
                backgroundColor: 'gray'
            });
            const section = Ti.UI.createListSection({
                items: [{
                        template: Ti.UI.LIST_ITEM_TEMPLATE_DEFAULT,
                        properties: {
                            title: 'backgroundSelectedColor',
                            backgroundSelectedColor: 'red'
                        }
                    },
                    {
                        template: Ti.UI.LIST_ITEM_TEMPLATE_DEFAULT,
                        properties: {
                            title: 'backgroundSelectedImage',
                            backgroundSelectedImage: 'https://garymathews.com/test.png'
                        }
                    },
                ]
            });
            const listView = Ti.UI.createListView({
                sections: [section]
            });
            win.add(listView);
            win.open();
        }
    }
].forEach(test => test.section.appendItems({
    title: {
        text: test.title
    },
    description: {
        text: test.description
    },
    properties: {
        test: test.test
    }
}));
win.add(listView);
win.open();
```

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-26304)

### CODE REVIEW

**Review by vijaysingh-axway — CHANGES_REQUESTED** [review]
Some minor changes required.

File: apidoc/Titanium/UI/ListItem.yml [hunk 1]
```diff
@@ -148,6 +148,37 @@ properties:
...
+  - name: backgroundSelectedImage
+    summary: Background image to render when the item is selected.
+    description: |
+        Must be a local resource.
+        On Android, clicking on ListItem's sub-views will not trigger this unless these views have 'touchEnabled' set to false.
+    type: String
+    since: 10.0.0
+    platforms: [android, iphone, ipad, macos]
+    accessors: false
+    deprecated:
```
[the remark this answers is not in the document]
Comment (garymathews) [reply apidoc/Titanium/UI/ListItem.yml]: Good catch!

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.

File: apidoc/Titanium/UI/ListItem.yml [hunk 2]
```diff
@@ -148,6 +148,34 @@ properties:
...
 
+  - name: backgroundSelectedColor
+    summary: Background color of the view, as a color name or hex triplet when item is selected.
+    description: |
+        On Android, clicking on ListItem's sub-views will not trigger this unless these views have 'touchEnabled' set to false.
+        For information about color values, see the "Colors" section of <Titanium.UI>.
+    type: [String, Titanium.UI.Color]
+    since: 10.0.0
+    platforms: [android, iphone, ipad, macos]
+    default: Transparent
```
Comment (sgtcoolguy) [inline apidoc/Titanium/UI/ListItem.yml:159]: is the default really `'transparent'` or `null`/`undefined`?

File: apidoc/Titanium/UI/ListItem.yml [hunk 3]
```diff
@@ -158,6 +186,9 @@ properties:
     platforms: [android, iphone, ipad, macos]
     default: Transparent
     accessors: false
+    deprecated:
+        since: "10.0.0"
+        notes: This property has been deprecated in favor of backgroundSelectedColor.
```
Comment (sgtcoolguy) [inline apidoc/Titanium/UI/ListItem.yml:191]:
```suggestion
        notes: This property has been deprecated in favor of `backgroundSelectedColor`.
```

File: apidoc/Titanium/UI/ListItem.yml [hunk 4]
```diff
@@ -168,13 +199,19 @@ properties:
     since: 3.2.0
     platforms: [android, iphone, ipad, macos]
     accessors: false
+    deprecated:
+        since: "10.0.0"
+        notes: This property has been deprecated in favor of backgroundSelectedImage.
```
Comment (sgtcoolguy) [inline apidoc/Titanium/UI/ListItem.yml:204]:
```suggestion
        notes: This property has been deprecated in favor of `backgroundSelectedImage`.
```

or 

```suggestion
        notes: This property has been deprecated in favor of <Titanium.UI.ListItem.backgroundSelectedImage>.
```

File: apidoc/Titanium/UI/ListItem.yml [hunk 5]
```diff
@@ -168,13 +199,19 @@ properties:
...
 
   - name: selectedBackgroundGradient
     summary: Background gradient to render when the item is selected.
     type: Gradient
     since: 3.2.0
     platforms: [iphone, ipad, macos]
     accessors: false
+    deprecated:
+        since: "10.0.0"
+        notes: This property has been deprecated in favor of backgroundSelectedGradient.
```
Comment (sgtcoolguy) [inline apidoc/Titanium/UI/ListItem.yml:214]:
```suggestion
        notes: This property has been deprecated in favor of `backgroundSelectedGradient`.
```

or

```suggestion
        notes: This property has been deprecated in favor of <Titanium.UI.ListItem.backgroundSelectedGradient>.
```

File: apidoc/Titanium/UI/TableViewRow.yml [hunk 6]
```diff
@@ -323,15 +323,33 @@ properties:
...
+    deprecated:
+        since: "10.0.0"
+        notes: This property has been deprecated in favor of backgroundSelectedColor.
 
   - name: selectedBackgroundImage
     summary: Background image to render when the row is selected.
     type: String
+    deprecated:
+        since: "10.0.0"
+        notes: This property has been deprecated in favor of backgroundSelectedImage.
```
Comment (sgtcoolguy) [inline apidoc/Titanium/UI/TableViewRow.yml:352]:
```suggestion
        notes: This property has been deprecated in favor of `backgroundSelectedImage`.
```

or

```suggestion
        notes: This property has been deprecated in favor of <Titanium.UI.TableViewRow.backgroundSelectedImage>.
```

**Comment by ssekhri** [comment]
FR Passed.
Verified on:
Mac OS: 10.15.4
SDK: 10.1.0.v20210309145621 
Appc CLI: 8.1.1
JDK: 11.0.6
Node: 12.16.1
Studio: 6.0.0.202005141803
Xcode: 12.3,
iOS simulator 14.3
Android emulator v11.0

**Pull request merged**